src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -321,9 +323,9 @@ public int GetHashCode(JsonSerializerOptions options)
                 hc.Add(options._writeIndented);
                 hc.Add(options._serializerContext);
 
-                foreach (JsonConverter converter in options.Converters)
+                for (int i = 0; i < options._converters.Count; i++)
```

Comment:
If you were to change:
https://github.com/dotnet/runtime/blob/9293aab2379d6a122d0cf5ed350356cb23e4d698/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ConverterList.cs#L72
to instead return `List<T>.Enumerator` (and then add an explicit implementation of the interface method), you could still foreach over the ConverterList without allocating.